ui/selectmenu.js

```diff
@@ -578,8 +576,10 @@ return $.widget( "ui.selectmenu", {
 	_toggleAttr: function() {
 		this.button.attr( "aria-expanded", this.isOpen );
 
-		this._toggleClass( this.button, "ui-selectmenu-button-closed", null, !this.isOpen )
-			._toggleClass( this.button, "ui-selectmenu-button-open", null, this.isOpen )
+		this._removeClass( this.button, "ui-selectmenu-button-" +
```

Comment:
Fantastic question that i thought i had a comment here for as this took me a while to figure out. 
You must ensure that classes are removed before they are added if your acting on the same element. 
Otherwise if they both have the same theme class it will un intentionally be removed after you add it.